Authenticator/Source/RootViewController.swift

```diff
@@ -201,6 +205,49 @@ extension RootViewController {
         )
         presentViewControllers([viewControllerA, viewControllerB])
     }
+
+    private func updateWithAuthViewModel(_ viewModel: AuthViewModel) {
+        if viewModel.enabled == currentViewModel.privacy.enabled {
+            return
+        }
+        if viewModel.enabled {
+            if authController == nil {
+                authController = UIViewController()
+                authController?.view.backgroundColor = UIColor.otpBackgroundColor
+                let button = UIButton(type: .roundedRect)
+                button.setTitleColor(UIColor.otpForegroundColor, for: .normal)
+                button.setTitle("Unlock", for: .normal)
+                button.addTarget(self, action: #selector(authChallenge), for: .touchUpInside)
+                button.sizeToFit()
+                authController?.view.addSubview(button)
+                button.center = authController!.view.center
+                authController?.modalPresentationStyle = .overFullScreen
+            }
+
+            guard let controller = authController else {
+                return
+            }
+            if let presented = presentedViewController {
+                presented.present(controller, animated: false)
+                return
+            } else {
+                present(controller, animated: false)
+            }
+        }
+        if !viewModel.enabled {
+            authController?.presentingViewController?.dismiss(animated: true)
+            authController = nil
+        }
+    }
+
+    @objc private func authChallenge() {
+        let context = LAContext()
+        context.evaluatePolicy(.deviceOwnerAuthentication, localizedReason: "LOLZ") { (reply, error) in
```

Comment:
This currently only triggers the passcode-screen, although Face ID is available. Per [docs](https://developer.apple.com/documentation/localauthentication/lapolicy/lapolicydeviceownerauthentication?language=objc) it should try Face ID / Touch ID first. Did you notice this as well?

Reply:
Strange! On my phone, it triggers FaceID.
Each app that wants to use FaceID shows a permission prompt the first time it tries to authenticate with FaceID. If permission is denied, it falls back to using a password. If you check in the system Settings.app, does Authenticator have permission to use FaceID?

<img src="https://user-images.githubusercontent.com/532638/57571230-3b35f480-73d9-11e9-89a6-c15a70d420b3.png" width="375pt">

Reply:
It does not prompt for this permission and is not visible in the settings but I can check again!

Having this pull request merged brings this project to a new level of privacy, so thanks everyone contributing to this! 🙂